timeseries/src/autogluon/timeseries/models/sktime/abstract_sktime.py

```diff
@@ -67,30 +67,32 @@ def __init__(
             hyperparameters=hyperparameters,
             **kwargs,
         )
-        self.skt_forecaster: Optional[BaseForecaster] = None
+        self.sktime_forecaster: Optional[BaseForecaster] = None
         self._fit_index: Optional[pd.Index] = None
 
-    def _get_skt_forecaster_args(self, args: dict, min_length: int, inferred_period: int = 1):
-        """Get arguments that will be passed to the sktime forecaster.
+    def _get_sktime_forecaster_init_args(
+        self, model_params: Dict[str, Any], min_length: int, inferred_period: int = 1
+    ):
+        """Get arguments that will be passed to the sktime forecaster at initialization.
 
-        These may be modified based on the length of training series and inferred seasonality.
+        model_params may be modified in-place based on the length of training series and inferred seasonality.
```

Comment:
oh, I missed this before. I believe this method should guarantee not to modify _in-place_. I also see in other places we `pop` from this dictionary. 

More generally, I would treat all arguments as `const` (in other languages) unless there's a strong reason not to do so. what do you think?

Reply:
you'll see in other places in the code we very often do `copy.deepcopy` of hyperparameter dicts to avoid this.

Reply:
Agreed, the new version fixes this. Now we call `self._get_model_params().copy()` inside `_get_sktime_forecaster_init_args` (instead of passing a copy of `model_params` to this function). No more in-place modifications :)